File: 3-longest-substring-without-repeating-characters/longest-substring-without-repeating-characters.java
class Solution {
    public int lengthOfLongestSubstring(String s) {
         int maxlen =0;
        
        for(int i =0;i<s.length();i++){
             HashSet<Character>set=new HashSet<>();
             for(int j = i ;j<s.length();j++){
                if(set.contains(s.charAt(j))){
                    break;
                }
                set.add(s.charAt(j));
                int len = j-i+1;
                maxlen=Math.max(len , maxlen);
             }
           
        }
        return maxlen;
        
    }
}

File: 3-longest-substring-without-repeating-characters/longest-substring-without-repeating-characters.cpp
class Solution {
public:
    int lengthOfLongestSubstring(string s) {
         int left =0;
         int ans =0;
         unordered_set<char>set;
         for(int i =0;i<s.size();i++){
            while(set.count(s[i])){
                set.erase(s[left]);
                left++;
            }
            set.insert(s[i]);
            int len = i-left+1;
            ans =max(ans,len);
            
             }
             return ans;
        
    }
};